src/Promitor.Core/Metrics/MeasuredMetric.cs

```diff
@@ -63,9 +63,11 @@ public static MeasuredMetric CreateForDimension(double? value, string dimensionN
         {
             Guard.NotNullOrWhitespace(dimensionName, nameof(dimensionName));
             Guard.NotNull(timeseries, nameof(timeseries));
-            Guard.For<ArgumentException>(() => timeseries.Metadatavalues.Any() == false);
+            
+            var dimensionMetadataValue = timeseries.Metadatavalues.Where(metadataValue => metadataValue.Name?.Value.Equals(dimensionName, StringComparison.InvariantCultureIgnoreCase) == true);            
+            Guard.For<ArgumentException>(() => dimensionMetadataValue.Count() == 0);
```

Comment:
Let's improve the handling a bit and throw a dedicated exception.
```suggestion
            var missingDimensionFound = timeseries.Metadatavalues.Any(metadataValue => metadataValue.Name?.Value.Equals(dimensionName, StringComparison.InvariantCultureIgnoreCase) == true);       
            if(missingDimensionFound)
            {
                throw new MissingDimensionException(dimensionName, timeseries);
            }
```